src/ol/interaction/Select.js

```diff
@@ -213,6 +213,12 @@ class Select extends Interaction {
      */
     this.hitTolerance_ = options.hitTolerance ? options.hitTolerance : 0;
 
+    /**
+     * @private
+     * @type {boolean}
+     */
+    this.userSpecifiedFeatureOverlay_ = !!options.featureOverlay;
```

Comment:
Personally I would be a bit more clear, e.q.   
`this.userSpecifiedFeatureOverlay_ = options.featureOverlay ? false : true;` 
but it has the same result. Even less obscure would be to do something like:  
```
this.userSpecifiedFeatureOverlay_ = typeof options.featureOverlay !== "undefined" 
                                    && options.featureOverlay !== null
```
Then the user would run into problems when he defines `options.featureOverlay` as something other than `null` or an actual `VectorLayer`, but that would be expected.

Reply:
Sorry for seeing this just now, otherwise the PR looks good!

Reply:
Thanks for the pointer. With so many different ways of doing things, it's always great to be introduced to a project's style so the code looks homogenous. I amended the code as follows:

```diff
diff --git a/src/ol/interaction/Select.js b/src/ol/interaction/Select.js
index 72b7217a4..1c62ff3dc 100644
--- a/src/ol/interaction/Select.js
+++ b/src/ol/interaction/Select.js
@@ -217,13 +217,14 @@ class Select extends Interaction {
      * @private
      * @type {boolean}
      */
-    this.userSpecifiedFeatureOverlay_ = !!options.featureOverlay;
+    this.userSpecifiedFeatureOverlay_ = typeof options.featureOverlay !== 'undefined' &&
+      options.featureOverlay !== null;
 
     /**
      * @private
      * @type {module:ol/layer/Vector}
      */
-    this.featureOverlay_ = options.featureOverlay ? options.featureOverlay : new VectorLayer({
+    this.featureOverlay_ = this.userSpecifiedFeatureOverlay_ ? options.featureOverlay : new VectorLayer({
       source: new VectorSource({
         useSpatialIndex: false,
         features: options.features,
```

Reply:
No problem. I'm not really an OpenLayers pro, but since I added the issue, I thought I could review your PR. I think you copy pasted my comment code? Because now the indentation looks a bit off (sorry.. 😉 ) If that's fixed I'll approve it!

Reply:
The indentation was actually a conscious choice, as I was trying to conform to this project's styling, for example see here:

https://github.com/openlayers/openlayers/blob/a94dff2c0635cce72f2ca6de32fe6805d26c0444/src/ol/interaction/Modify.js#L238-L243

I don't have a problem with indentation and can go either way, but would like to do that which is idiomatic to this project. Maybe a core developer can help us here.